File: frontend/src/types/book.ts
export interface Book {
  id: number;
  title: string;
  isbn: string;
  publishedYear: number;
  coverImageUrl?: string | null;
  coverImagePublicId?: string | null;
  publisherId: number;
  publisherName: string;
  categoryId: number;
  categoryName: string;
  authorNames: string[];
}

export interface CreateBookRequest {
  title: string;
  isbn: string;
  publishedYear: number;
  coverImageUrl?: string | null;
  coverImagePublicId?: string | null;
  publisherId: number;
  categoryId: number;
  authorIds: number[];
}

export type UpdateBookRequest = CreateBookRequest;

File: backend/Domain/Entities/Book.cs
using Domain.Common;

namespace Domain.Entities;

public class Book : BaseEntity
{
    public string Title { get; set; } = string.Empty;
    public string Isbn { get; set; } = string.Empty;
    public int PublishedYear { get; set; }
    public string? CoverImageUrl { get; set; }
    public string? CoverImagePublicId { get; set; }

    public int PublisherId { get; set; }
    public Publisher? Publisher { get; set; }

    public int CategoryId { get; set; }
    public Category? Category { get; set; }

    public ICollection<BookAuthor> BookAuthors { get; set; } = new List<BookAuthor>();
    public ICollection<BookCopy> BookCopies { get; set; } = new List<BookCopy>();
}
